resource_estimator/src/estimates/physical_estimation/estimate_without_restrictions.rs

```diff
@@ -107,7 +107,10 @@ impl<
                 ));
             }
 
-            num_cycles = max_allowed_num_cycles_for_code_parameter + 1;
+            num_cycles = std::cmp::max(
+                max_allowed_num_cycles_for_code_parameter + 1,
+                num_cycles + 1,
```

Comment:
I see that num_cycles is now guaranteed to increase. But how does it guarantee that loop eventually exits?